### ISSUE TITLE
Error when using check_archived_root for a BTreeSet

### ISSUE THREAD

**Issue by rklaehn** [rkyv/rkyv#208]
I am learning the library. I wrote a few simple examples to get the hang of it. I disovered this when trying to roundtrip serialize an empty `BTreeSet<u8>`:

```rust
    let t: BTreeSet<u8> = btreeset!{};
    let mut serializer = AllocSerializer::<256>::default();
    serializer.serialize_value(&t).unwrap();
    let bytes = serializer.into_serializer().into_inner();
    println!("bts: {}", hex::encode(bytes.as_ref()));
    let archived = check_archived_root::<BTreeSet<u8>>(&bytes).unwrap();
    let deserialized: BTreeSet<u8> = archived.deserialize(&mut Infallible).unwrap();
```

gives

```
bts: 00000000fcffffff
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: CheckBytesError(TupleStructCheckError { field_index: 0, inner: ContextError(ArchiveError(Overrun { ptr: 0x562d2f61ece0, size: 12, range: 0x562d2f61ece0..0x562d2f61ece8 })) })', src/main.rs:107:64
```

for a non-empty `BTreeSet<u8>`, it works fine:

```rust
    let t: BTreeSet<u8> = btreeset!{0u8};
    let mut serializer = AllocSerializer::<256>::default();
    serializer.serialize_value(&t).unwrap();
    let bytes = serializer.into_serializer().into_inner();
    println!("bts: {}", hex::encode(bytes.as_ref()));
    let archived = check_archived_root::<BTreeSet<u8>>(&bytes).unwrap();
    let deserialized: BTreeSet<u8> = archived.deserialize(&mut Infallible).unwrap();
```

gives

```
bts: 0000000000000000010000000000000001000000ecffffff
```

Is this a bug, or am I doing something wrong?

**Comment by djkoloski** [comment]
I think this is probably a bug, the validation logic has a tendency to be overly restrictive at times. I'll take a look at this and see if I can reproduce the issue.

**Comment by djkoloski** [comment]
Fixed by e680216454351ae20bb51861412d832c98bccff3, this will be in the next release.

**Comment by rklaehn** [comment]
Yes, I noticed that it does work when using the unsafe method. Thanks a lot for fixing this so quickly!
